backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -123,25 +154,37 @@ public void deleteOrganization(String externalId) {
     }
 
     // returns the external ID of the organization the specified user belongs to
-    public String getOrganizationExternalIdForUser(String username) {
+    public Optional<OrganizationRoleClaims> getOrganizationRolesForUser(String username) {
+
         UserList users = _client.listUsers(username, null, null, null, null);
         if (!users.iterator().hasNext()) {
             throw new IllegalGraphqlArgumentException("Cannot get org external ID for nonexistent user");
         }
         User user = users.single();
 
+        Set<String> orgExternalIds = new HashSet<>();
+        Set<OrganizationRole> roles = new HashSet<>();
+
         for (Group g : user.listGroups()) {
             String groupName = g.getProfile().getName();
-            // We assume that a user is only a member of one user group
-            if (g.getType() == GroupType.OKTA_GROUP &&
-                    groupName.startsWith(_rolePrefix) &&
-                    groupName.endsWith(generateRoleSuffix(OrganizationRole.USER))) {
-                return getOrganizationExternalIdFromGroupName(groupName, 
-                                                              OrganizationRole.USER);
+            for (OrganizationRole role : OrganizationRole.values()) {
```

Comment:
I think this code path and the one in OrganizationExtractor should not both exist in the long term, but I think this is in keeping with the other question about longer-term business rules for inactive orgs, and also potentially something we're going to refactor when we do V2 access control anyway.

Reply:
Yeah, and this is designed specifically for users that are *not* the current user